include/ffi.h.in

```diff
@@ -315,7 +315,12 @@ ffi_prep_closure (ffi_closure*,
 		  ffi_cif *,
 		  void (*fun)(ffi_cif*,void*,void**,void*),
 		  void *user_data)
-  __attribute__((deprecated ("use ffi_prep_closure_loc instead")));
+#if defined(__GNUC__) && (((__GNUC__ * 100) + __GNUC_MINOR__) >= 450)
```

Comment:
I think here `__GNUC_MINOR__` will be `5` for gcc 4.5, so the resulting value would be 405, not 450.  So I believe this needs a minor tweak one way or the other.